jakarta-eclipse/src/org/jakarta/codeAction/HttpServletQuickFix.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.jakarta.codeAction;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.internal.corext.dom.Bindings;
+import org.eclipse.jdt.internal.core.manipulation.util.BasicElementLabels;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.ltk.core.refactoring.NullChange;
+
+import org.jakarta.jdt.ServletConstants;
+import org.jakarta.lsp4e.Activator;
+import org.jakarta.codeActionProposal.ChangeCorrectionProposal;
+
+public class HttpServletQuickFix implements IJavaCodeActionParticipant{
+	@Override
+	public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+			IProgressMonitor monitor) throws CoreException {
+		ASTNode node = context.getCoveredNode();
+		ITypeBinding parentType = Bindings.getBindingOfParentType(node);
+		if (parentType != null) {
+			List<CodeAction> codeActions = new ArrayList<>();
+			// Create code action
+			// interface
+			final String TITLE_MESSAGE = "Let ''{0}'' extend ''{1}''";
+			String args[] = { BasicElementLabels.getJavaElementName(parentType.getName()),
+					BasicElementLabels.getJavaElementName(ServletConstants.HTTP_SERVLET) };
+			ChangeCorrectionProposal proposal = new ChangeCorrectionProposal(parentType.getName(),
+					CodeActionKind.QuickFix, new NullChange(""));
+//			ChangeCorrectionProposal proposal = new ImplementInterfaceProposal(context.getCompilationUnit(), parentType,
+//					context.getASTRoot(), ServletConstants.HTTP_SERVLET);
+			CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
+			proposal.setDisplayName(MessageFormat.format(TITLE_MESSAGE, args));
+			codeActions.add(codeAction);
+			return codeActions;
+		}
+		return null;
```

Comment:
I used nullChange to initiate the change class for now. 

It seems not working still, what is the correct way to create a change properly?

Reply:
See my message on slack